build/pipeline-pullrequest.yaml

```diff
@@ -0,0 +1,95 @@
+
+# PR builds should be triggered on every branch except master (so we don't trigger again after a merge)
+trigger:
+  branches:
+    include:
+    - '*'
+    exclude:
+    - master
+
+pool:
+  name: Hosted VS2017
+  demands:
+  - msbuild
+  - visualstudio
+
+variables:
+  BuildPlatform: 'any cpu'
+  BuildConfiguration: 'debug'
+
+steps:
+- task: AzureKeyVault@1
+  displayName: 'Azure Key Vault: BuildAutomation'
+  inputs:
+    azureSubscription: 'MSID MSAL.NET Automation KeyVault'
+    KeyVaultName: BuildAutomation
+    SecretsFilter: 'AzureADIdentityDivisionTestAgent, RSATestCertDotNet'
+
+- powershell: |
+   $kvSecretBytes = [System.Convert]::FromBase64String('$(AzureADIdentityDivisionTestAgent)')
+   $certCollection = New-Object System.Security.Cryptography.X509Certificates.X509Certificate2Collection
+   $certCollection.Import($kvSecretBytes, $null, [System.Security.Cryptography.X509Certificates.X509KeyStorageFlags]::Exportable)
+
+   $protectedCertificateBytes = $certCollection.Export([System.Security.Cryptography.X509Certificates.X509ContentType]::Pkcs12)
+   $pfxPath = '$(Build.SourcesDirectory)' + "\TestCert.pfx"
+   [System.IO.File]::WriteAllBytes($pfxPath, $protectedCertificateBytes)
+
+   Import-PfxCertificate -FilePath $pfxPath -CertStoreLocation Cert:\LocalMachine\My
```

Comment:
>Cert:\LocalMachine\My [](start = 63, length = 21)

Should the certs be in the user store instead? Also seems like the certs are not removed again when the build is complete, should they be?

Reply:
Ive seen instances where the cert cannot be found if in the user store for some unknown reason so I usually put it in local machine. So far, tests haven't had an issue finding them there. Also, this will be running on a hosted agent which will be wiped after we finish using it so I think its ok to leave the certs there.